### PR TITLE
Refactor log flushing and log type handling

### PR DESCRIPTION
**Pull request by danrodrig**
- Dedupe log flushing
- Add helpers for log types

### CODE REVIEW

File: src/logging.c [hunk 1]
```diff
@@ -376,32 +409,18 @@ rstrnt_upload_log (const RstrntTask    *task,
...
-            data_file = data->harness_log_data->file;
-            log_path = LOG_PATH_HARNESS;
-        }
-        break;
-    }
+    log_path = rstrnt_log_type_get_path (type);
+    log_data = rstrnt_task_log_get_data (data, type);
 
-    path = g_file_get_path (data_file);
+    path = g_file_get_path (log_data->file);
```
Comment (cbouchar) [inline src/logging.c:423]: I think this is a local variable as I'm not seeing it absorbed anywhere.  If that's true, then I believe path needs to be g_free'd according to API doc.

File: src/logging.c [hunk 2]
```diff
@@ -251,13 +251,30 @@ rstrnt_on_log_uploaded (SoupSession *session,
...
+    g_autofree gchar   *path = NULL;
+    g_autoptr (GError)  error = NULL;
+
+    stream = G_OUTPUT_STREAM (log_data->output_stream);
+
+    if (g_output_stream_flush (stream, cancellable, &error)
+        || G_IO_ERROR_CANCELLED == error->code)
+        return;
+
+    path = g_file_get_path (log_data->file);
```
Comment (cbouchar) [inline src/logging.c:268]: API doc says to g_free 'path' when no longer needed.

Comment (danrodrig) [reply src/logging.c:268]: Line 259, `g_autofree gchar   *path = NULL;`

And valgrind checks would have complained about that.

File: src/logging.c (hunk 1 above)
Comment (danrodrig) [reply src/logging.c:423]: Line 405, ` g_autofree char *path = NULL;` same as when the code was introduced.

File: src/logging.c (hunk 2 above)
Comment (cbouchar) [reply src/logging.c:268]: I looked for g_autofree but saw log_path definition instead.We're good.

File: src/logging.c (hunk 1 above)
Comment (cbouchar) [reply src/logging.c:423]: same comments as earlier.

**Review by cbouchar — APPROVED** [review]

File: src/logging.c (hunk 2 above)
Comment (danrodrig) [reply src/logging.c:268]: > I looked for g_autofree but saw log_path definition instead.We're good.

There is no `log_path` in this function.

**Pull request merged**